File: AngularClient/src/app/models/expense.ts
import { User } from './user';
import { Category } from './category';
export class Expense {
    expenseId: number;
    userId: number;
    categoryId: number;
    date?: Date;
    amount?: number;
    title?: string;
    description?: string;
    vendor?: string;
    isBusinessExpense?: boolean;

    category?: Category;
    user?: User;

    constructor() {
        this.userId = 0;
        this.categoryId = 0;
        this.description = '';
        this.title = '';
        this.amount = 0;
        this.date = new Date();
        this.vendor = '';
        this.isBusinessExpense = false;
    }
}

File: BudgetService/BudgetService.API/Models/Dto/Expense.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace BudgetService.API.Models.Dto
{
    [Table("BudgetItem")]
    public class Expense
    {
        [Column("BudgetItemId")]
        public int ExpenseId { get; set; }
        public int UserId { get; set; }
        public int CategoryId { get; set; }
        public DateTime? Date { get; set; }
        public decimal? Amount { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Vendor { get; set; }
        public bool? IsBusinessExpense { get; set; }


    }
}